project1/application.py

```diff
@@ -116,3 +120,47 @@ def userHome(Username):
     if Username in session:
         return render_template("userDetails.html", username=Username, message="Successfully logged in.", heading="Welcome back")
     return redirect(url_for('index'))
+
+
+
+@app.route("/search",methods=["POST","GET"])
+def search():
+	if request.method == "GET":
+		return render_template("search.html")
+	else:
+		result = request.form.get("search")
+		result = '%'+result+'%'
+		search_result = Books.query.filter(or_(Books.tittle.ilike(result), Books.author.ilike(result), Books.isbn.ilike(result),Books.year.ilike(result))).all()
+		return render_template("search.html", books=search_result)
+
+
+@app.route("/bookpage/<username>/<isbn>", methods = ["POST","GET"])
+def bookpage(username, isbn):
+	user1 = username
+
+	if user1 in session:
+		user_reviews = reviewRate.query.filter_by(isbn = book_isbn).all()
+		if request.method == "GET":
+			revie = reviewRate.query.filter(reviewRate.isbn.like(bookisbn),reviewRate.username.like(user1)).first()
+
+			user_reviews = reviewRate.query.filter_by(isbn=bookisbn).all()
+
+			if rev is None:
+				return render_template("bookpage.html",book=book,res=res,revie = user_reviews,username = user1)
+			return render_template("bookpage.html",book=book,message="You already given review!",revie = user_reviews,res=res,property="none",username=user1)
+
+		else:
+			rating = request.form.get("rating")
+			reviews = request.form.get("review")
+
+			isbn = book_isbn
+			username = user1
+
```

Comment:
How are you making sure that the user should be able to post exactly one review for one particular book?

Reply:
we are implementing that by checking if a review is given or not by if condition. And then if review given, we making the review block to hide using  "property- none". u can see it in lines 155